FirebaseAdmin/FirebaseAdmin/Auth/Providers/OidcProviderConfig.cs

```diff
@@ -63,13 +65,60 @@ internal OidcProviderConfig(Request request)
         /// </summary>
         public string Issuer { get; }
 
+        /// <summary>
+        /// Gets the Client Secret used to verify code based response types.
+        /// </summary>
+        public string ClientSecret { get; }
+
+        /// <summary>
+        /// Gets what response types are being used for this OIDC config.
+        /// </summary>
+        public ResponseTypeArgs ResponseType { get; }
+
+        /// <summary>
+        /// Represents the response types beinf used for this OIDC config.
+        /// </summary>
+        public class ResponseTypeArgs
+        {
+            internal ResponseTypeArgs(ResponseTypeJson request)
+            {
+                this.Code = request.Code;
+                this.IDToken = request.IDToken;
+            }
+
+            /// <summary>
+            /// Gets a value indicating whether this OIDC provider uses a code based response type.
+            /// </summary>
+            public bool? Code { get; }
+
+            /// <summary>
+            /// Gets a value indicating whether this OIDC provider uses an ID-token based response type.
+            /// </summary>
+            public bool? IDToken { get; }
+        }
+
+        internal sealed class ResponseTypeJson
```

Comment:
May be call this `ResponseTypeMap` or `ResponseTypeInfo`.

Reply:
Done